aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -417,7 +417,11 @@ Tensor linspace(
     int64_t steps,
     const TensorOptions& options) {
   TORCH_CHECK(steps >= 0, "number of steps must be non-negative");
-  Tensor result = at::empty({steps}, options);
+  auto result_options = options;
```

Comment:
also we should error out when the input dtype is not complex, but the start and end are. this is tricky here because the options would by default have a floating point dtype.
cc. @mruberry

Reply:
Which input dtype, the function only takes `start`, `stop` and `steps` as the main input to compute.

Reply:
Understood now, please check updated the comment below this line and let me know if there is a graceful way to do it.

Reply:
yeah it's a kwarg https://pytorch.org/docs/stable/torch.html#torch.linspace

Reply:
Is there a way to know if it was defaulted or user specified?

Reply:
Unfortunately I don't think there is, @bhosmer to verify. In a similar circumstance I had to implement a custom dispatch path (see https://github.com/pytorch/pytorch/pull/34709).

Reply:
Oh. Thanks for the reference. Will have a look.

Reply:
thanks @mruberry I think we can use a similar approach here as well. in fact, it might be useful to generalize the `infer_full_options` to make it usable at other places like here as well.

Reply:
I am not sure if it can be done in a very generic way as ops do differ in their behaviour of inferring the type like the inputs that they consider, the types they default to.
(However, I don't know much about the same so my assumption above might be wrong).

For now have put the inference code as a function similar to `infer_full_options`.